libc/sysdeps/linux/arc/sigrestorer.S

```diff
@@ -22,4 +22,5 @@
 __default_rt_sa_restorer:
 	mov r8, __NR_rt_sigreturn
 	ARC_TRAP_INSN
+	j_s     [blink]
```

Comment:
Yes I have noticed this in the past and tried this but at the time I had more failures 
It is likely this was due to missing pairing libgcc fix.

Reply:
In fact this jump is not needed, at least for relatively new kernels. But to minimize the impact on backwards compatibility, it makes sense to add it here rather than remove it in glibc and update signal handler unwinder in libgcc.

Reply:
Right that is understood, and in my original trial to add this I had a comment which called that out.

[ __default_rt_sa_restorer:
+       cfi_startproc
+
        mov r8, __NR_rt_sigreturn
        ARC_TRAP_INSN
+       ; the trap never returns, but this j keeps unwind info sane
+       j_s [blink]
+
+       cfi_endproc](url)

And I'm not suggesting this is removed from glibc as it will certainly break unwinder detection logic.

This change in theory is good. When I tried adding it - I regressed uClibc testsuite as I was likely missing your libgcc fix (which is great sleuthing BTW - good job).

We just need to carefully document the baseline and record what fixes what - as this fix alone will regress w/o libgcc fix.

Reply:
If you really want to tackle uClibc failures I suggest you also enable cfi pseudo ops support in ARC uClibc port.

+#define __UCLIBC_HAVE_ASM_CFI_DIRECTIVES__

Reply:
Alternatively, you can change the detection pattern in libgcc (TRAP_S_J_S_BLINK) at [line 66](https://github.com/foss-for-synopsys-dwc-arc-processors/gcc/blob/d9a09c0c9afec472e2914b972eefced52947b3b3/libgcc/config/arc/linux-unwind.h#L66) and [line 70](https://github.com/foss-for-synopsys-dwc-arc-processors/gcc/blob/d9a09c0c9afec472e2914b972eefced52947b3b3/libgcc/config/arc/linux-unwind.h#L70). But, probably this patch is the cleanest.